COBOL Programming Course #4 - Testing/COBOL Programming Course #4 - Testing.md

```diff
@@ -188,178 +186,664 @@ Even more damaging could be the impact of being unable to deliver function rapid
 
-The key point is that anything that lends itself to automation should be automated, to allow time for these other activities where automation is genuinely not appropriate.  The fact that something is quite difficult to automate should not be used as an excuse for failing to automate it.
 
- 
-### Managing the pipeline when test cases fail
 
-As noted above, if the tests are automated, but checking or reacting to their results is not, then that is only half of the story.  When building a CI/CD pipeline, thought should be given to what happens when test cases fail. 
+### Test suites and test cases
 
- 
+TESTSUITE - Provides a description for a series of test cases. The description is echoed in the output from the test run.
+TESTCASE - identifies a test case. The description is echoed in the output of the test run.
 
-The diagnostics that will reveal the cause of the failure should be collected in a known location and made available in an easy way to the engineer who will investigate the problem.  The software change that caused the failure should be easily identifiable, and its progress through the pipeline should be halted, or reversed.  Other software changes that have not caused the failure should be able to continue unimpeded, unless they are only viable with the failing change, which should also be something the pipeline can detect and act upon.  Finally, it is possible that the failure could be due to a problem with the test or the test environment, and those possibilities must also be easy to detect and investigate.
+A program can have multiple test suites, and each test suite can have multiple test cases.
 
- 
-## Testing on z/OS
+The typical automated check follows these steps:
 
-Many organizations have chosen IBM Z for the most critical aspects of their businesses, especially in industries including banking, insurance, and retail, where disruptions cannot be tolerated.  This means that in addition to relying on the inherent reliability, security and resilience of the platform, such companies also carry out extensive testing before introducing any change.
+Arrange: Establish the preconditions for the test case.
 
+Act: Execute the code under examination.
 
-### Importance of testing on z/OS
+Assert: Validate the anticipated outcome of the test
```

Comment:
We should probably change this to bullet point as well so that it will be easier to read